### PR TITLE
Remove --int-literals option

### PR DESCRIPTION
**Pull request by ihnorton**
Did I miss anything? ref #9567

### CODE REVIEW

**Comment by tkelman** [comment]
Is it mentioned on the man page or in the docs? getting-started and I think one or two other places mention (some of) command-line args but have gotten out of date.

**Comment by ihnorton** [comment]
Was in the first section of manual - thanks.

**Comment by tkelman** [comment]
also https://github.com/JuliaLang/julia/blob/96d97ca582520040f41b668ceed06b045bd5e280/doc/man/julia.1#L123-L125

File: src/ast.c [hunk 1]
```diff
@@ -245,20 +245,6 @@ static jl_value_t *scm_to_julia_(value_t e, int eo)
...
-            jl_compileropts.int_literals!=64
-#endif
-            ) {
-            if (i64 > (int64_t)S32_MAX || i64 < (int64_t)S32_MIN)
-                return (jl_value_t*)jl_box_int64(i64);
-            return (jl_value_t*)jl_box_int32((int32_t)i64);
-        }
-        else {
-            return (jl_value_t*)jl_box_int64(i64);
-        }
```
Comment (JeffBezanson) [inline src/ast.c]: Whoa, deleted way too much here. All this code should remain, just with different platform ifdefs.

**Comment by JeffBezanson** [comment]
Also need to change the `JLCompilerOpts` struct in inference.jl.

File: src/ast.c [hunk 2]
```diff
@@ -245,20 +245,6 @@ static jl_value_t *scm_to_julia_(value_t e, int eo)
...
             i64 = conv_to_int64(cp_data(cp), nt);
         }
-        if (
-#ifdef _P64
-            jl_compileropts.int_literals==32
-#else
-            jl_compileropts.int_literals!=64
-#endif
-            ) {
-            if (i64 > (int64_t)S32_MAX || i64 < (int64_t)S32_MIN)
```
Comment (JeffBezanson) [inline src/ast.c]: This should always be used on 32-bit.

File: src/ast.c [hunk 3]
```diff
@@ -245,20 +245,6 @@ static jl_value_t *scm_to_julia_(value_t e, int eo)
...
-                return (jl_value_t*)jl_box_int64(i64);
-            return (jl_value_t*)jl_box_int32((int32_t)i64);
-        }
-        else {
-            return (jl_value_t*)jl_box_int64(i64);
```
Comment (JeffBezanson) [inline src/ast.c]: This should always be used on 64-bit.

File: src/ast.c (hunk 1 above)
Comment (ihnorton) [inline src/ast.c]: Oops :)

**Comment by JeffBezanson** [comment]
Also the option help text in repl.c.

File: ui/repl.c [hunk 4]
```diff
@@ -92,11 +92,10 @@ void parse_opts(int *argcp, char ***argvp)
         { "track-allocation",required_argument, 0, 'm' },
         { "check-bounds",  required_argument, 0, 300 },
         { "optimize",      no_argument,       0, 'O' },
-        { "int-literals",  required_argument, 0, 301 },
-        { "dump-bitcode",  required_argument, 0, 302 },
-        { "compile",       required_argument, 0, 303 },
-        { "depwarn",       required_argument, 0, 304 },
-        { "inline",        required_argument, 0, 305 },
+	    { "dump-bitcode",  required_argument, 0, 301 },
```
Comment (jiahao) [inline ui/repl.c]: Ref: #9482

**Comment by ihnorton** [comment]
Rebased.

**Comment by ihnorton** [comment]
If there are no objections I'll merge this later today.

**Comment by PallHaraldsson** [comment]
I kind of liked having this - for testing - mostly just saw it was broken.. 32-bit VM should do.. and then Julia code should work for sure (as in physical).

And another thing: My switchable literal type idea was shot down. For integer (or possibly for other types, say, strings - but not Decimal Float-literals..), should switching on runtime be possible as an alternative?

**Pull request merged**